azurerm/resource_arm_analysis_services_server.go

```diff
@@ -0,0 +1,449 @@
+
+			"location": azure.SchemaLocation(),
+
+			"sku": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"D1",
+					"B1",
+					"B2",
+					"S0",
+					"S1",
+					"S2",
+					"S4",
+					"S8",
+					"S9",
+				}, false),
+			},
+
+			"admin_users": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			//"backup_blob_container_uri": {
+			//	Type:         schema.TypeString,
+			//	Optional:     true,
+			//	Sensitive:    true,
+			//	ValidateFunc: validate.URLIsHTTPOrHTTPS,
+			//},
+
+			"enable_power_bi_service": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+
+			"ipv4_firewall_rule": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"range_start": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validate.IPv4Address,
+						},
+						"range_end": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validate.IPv4Address,
+						},
+					},
+				},
+			},
+
+			"querypool_connection_mode": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validateQuerypoolConnectionMode(),
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceArmAnalysisServicesServerCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).analysisServicesServerClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for Azure ARM Analysis Services Server creation.")
+
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+
+	if requireResourcesToBeImported && d.IsNewResource() {
+		server, err := client.GetDetails(ctx, resGroup, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(server.Response) {
+				return fmt.Errorf("Error checking for presence of existing Analysis Services Server %q (Resource Group %q): %s", name, resGroup, err)
+			}
+		}
+
+		if server.ID != nil && *server.ID != "" {
+			return tf.ImportAsExistsError("azurerm_analysis_services_server", *server.ID)
+		}
+	}
+
+	sku := d.Get("sku").(string)
+	location := azure.NormalizeLocation(d.Get("location").(string))
+
+	serverProperties := expandAnalysisServicesServerProperties(d)
+
+	if fwSettings := serverProperties.IPV4FirewallSettings; len(*fwSettings.FirewallRules) > 0 && fwSettings.EnablePowerBIService == nil {
+		return fmt.Errorf("`enable_power_bi_service` must be set if there is at least one firewall rule")
+	}
+
+	tags := d.Get("tags").(map[string]interface{})
+
+	analysisServicesServer := analysisservices.Server{
+		Name:             &name,
+		Location:         &location,
+		Sku:              expandAnalysisServicesServerSku(sku),
```

Comment:
Given this is just .a string i don't think we need an expand function here

Reply:
removed (in case you mean we can do it inline)